controllers/argocd/status.go

```diff
@@ -250,16 +252,33 @@ func (r *ReconcileArgoCD) reconcileStatusNotifications(cr *argoprojv1a1.ArgoCD)
 func (r *ReconcileArgoCD) reconcileStatusHost(cr *argoprojv1a1.ArgoCD) error {
 	cr.Status.Host = ""
 	cr.Status.Phase = "Available"
-	if cr.Spec.Server.Route.Enabled {
-		if !IsRouteAPIAvailable() {
-			log.Info("Routes not available in non-OpenShift environments, please use Ingresses instead")
-			return nil
-		}
```

Comment:
We should preserve the behavior that when the operator is deployed on a non-openshift cluster and enabling route, the above message is logged.

Reply:
I initially added it but during the code review, @jaideepr97 and I realized that it is a redundant piece of code.

https://github.com/argoproj-labs/argocd-operator/blob/2090abd831d1600ad31f81d71dfc7cff4652f9c0/controllers/argocd/util.go#L787

Reply:
Also, it looks out of the place to log this error message during statusReconciliation after all the processing is done.